cns/restserver/internalapi.go

```diff
@@ -143,6 +146,68 @@ func (service *HTTPRestService) SyncNodeStatus(dncEP, infraVnet, nodeID string,
 	return
 }
 
+// SyncHostNCVersion will check NC version from NMAgent and save it as host NC version in container status.
+// If NMAgent NC version got updated, CNS will refresh the pending programming IP status.
+func (service *HTTPRestService) SyncHostNCVersion(ctx context.Context, channelMode string, lastUpdatedHostNCTimeStamp time.Time, forceMarkIPAvailableTimeRange time.Duration) {
```

Comment:
I highly recommend to move this to a separate file (like IPAMPoolMonitor). we only add apis called by internal cns client in this file.

Reply:
This file is not for internal cns call from begining. It's AKS start using it for internal CNS call. There are 6 functions in internalapi.go, only 2 of them are called by cns client. CreateOrUpdateNetworkContainerInternal and ReconcileNCState are called by cns client. The rest of 4 are not called by cns client.

Reply:
Could I remain the function here for this version? 
I can do moving logic to separate file for next PR and we can move all the related logic, including what managed DNC use in a separate file.